network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py

```diff
@@ -35,9 +36,16 @@ class RecentBlogEntries(blocks.StructBlock):
         help_text="Optional divider below content block.",
     )
 
-    # TODO: add in validation so that if there are no tags or topic
-    #       filled in we don't allow the page to be saved, with a wagtail
-    #       error indication what's wrong.
+    def clean(self, value):
+        result = super().clean(value)
+
+        if result["tag_filter"] and result["topic_filter"]:
+            raise exceptions.ValidationError("Please provide either a Tag or a Topic, not both", code="invalid")
```

Comment:
**Kudos** Very cool to see the use of the `code` parameter for `ValidationError`. I was actually not aware that this existed! 

I don't think that this has been used anywhere else in the code base, but I can see this being very useful in the future, especially when doing things such as testing custom validation and logic for our models. 

Going to take note of this myself for future use 🔖